runtime/vm/stringhelpers.cpp

```diff
@@ -258,131 +258,67 @@ compareStringToUTF8(J9VMThread *vmThread, j9object_t string, UDATA translateDots
 }
 
 #if !defined (J9VM_OUT_OF_PROCESS)
-
 /**
- * Copy certain number of Unicode characters from an offset into a Unicode character array to a UTF8 data buffer.
+ * Copy a Unicode String to a UTF8 data buffer with NULL termination. This function makes an assumption that the
+ * caller has guaranteed the buffer is of the appropriate size to fit the encoded UTF8 string.
```

Comment:
Pls specify the buffer size according to incoming string which can guaranteed it fits the encoded string.

Reply:
Doesn't this comment say exactly this? We give responsibility to the caller to provide us a buffer for the string which is guaranteed to fit any encoding of the specified length.